subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -360,6 +360,38 @@ static struct bass_recv_state_internal *get_free_recv_state(void)
 		}
 	}
 
+	LOG_DBG("All receive states occupied");
+
+#if defined(CONFIG_BT_BAP_SCAN_DELEGATOR_OVERWRITE_RECV_STATE)
```

Comment:
Does this intentionally not take age into account? 

Since this is starting from the same index, then the following will happen

1. Server has 3 receive states
2. Some clients or the server has added 3 receive states, so that all states are used, but none are PA nor BIG synced (none active)
3. A client then wants to a new source; this overwrites recv_state[0]
4. Another client (or the server) at the same time wants to do the same, but then that will overwrite recv_state[0]) if it hasn't synced to anything yet

For step 4, it would make more sense to overwrite [1] or [2] as they are less recently updated. 

I strongly suggest that we implement this feature in the same way that CONFIG_BT_KEYS_OVERWRITE_OLDEST is.